lib/snipr/process_signaller.rb

```diff
@@ -132,14 +132,23 @@ def dry_run
       @dry_run = true
     end
 
+    ##
+    # Ensure that the parent PID is still accurate
+    def ppid_matches?(process)
+      # PPID is the 4th item under /proc/pid/stat (whitespace-delimited)
+      process.ppid.to_i == File.read("/proc/#{process.pid}/stat").split[3].to_i
+    end
+
     private
     def signal_process(process)
       @before_signal.call(@signal, process)
       unless @dry_run
         if @target_parent
-          Process.kill(@signal, process.ppid)
+          Process.kill(@signal, process.ppid) if ppid_matches?(process)
```

Comment:
Why not also switch to using `pkill` here?

Reply:
oh, right, you don't know the parent pid's parent, nevermind

Reply:
Because the only thing we really know is the PPID itself and `pkill` can't search on a specific PID. Instead, we could use `kill`, but I'm not sure what would be gained by doing that.